starboard/nplb/posix_compliance/posix_realpath_test.cc

```diff
@@ -0,0 +1,240 @@
+// Copyright 2025 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+ The following error conditions are not tested due to the difficulty of
+ reliably triggering them in a portable unit testing environment:
+ - EACCES (permission errors)
+ - EIO (I/O errors)
+*/
+
+#include <errno.h>
+#include <limits.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "starboard/nplb/file_helpers.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace nplb {
+namespace {
+
+const mode_t kUserRwx = S_IRWXU;
+
+// Test fixture for realpath tests. It sets up a temporary directory with a
+// predefined structure of files, subdirectories, and symbolic links.
+class PosixRealpathTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    char template_name[] = "/tmp/realpath_test_XXXXXX";
+    char* dir_name = mkdtemp(template_name);
+    ASSERT_NE(dir_name, nullptr) << "mkdtemp failed: " << strerror(errno);
+    test_dir_ = dir_name;
+
+    // Create a known file/directory structure for tests:
+    // test_dir_
+    // |-- file1
+    // |-- subdir1
+    // |   `-- file2
+    // |-- symlink_to_file1 -> (absolute path to file1)
+    // |-- symlink_to_subdir1 -> (absolute path to subdir1)
+    // |-- dangling_symlink -> "non_existent"
+    // |-- loop_a -> "loop_b"
+    // `-- loop_b -> "loop_a"
+
+    // Create file1
+    std::string file1_path = test_dir_ + "/file1";
+    int fd = open(file1_path.c_str(), O_CREAT | O_WRONLY, 0600);
+    ASSERT_NE(fd, -1);
+    ASSERT_EQ(close(fd), 0);
+
+    // Create subdir1 and file2
+    std::string subdir1_path = test_dir_ + "/subdir1";
+    ASSERT_EQ(mkdir(subdir1_path.c_str(), kUserRwx), 0);
+    std::string file2_path = subdir1_path + "/file2";
+    fd = open(file2_path.c_str(), O_CREAT | O_WRONLY, 0600);
+    ASSERT_NE(fd, -1);
+    ASSERT_EQ(close(fd), 0);
+
+    ASSERT_EQ(
+        symlink(file1_path.c_str(), (test_dir_ + "/symlink_to_file1").c_str()),
+        0);
+    ASSERT_EQ(symlink(subdir1_path.c_str(),
+                      (test_dir_ + "/symlink_to_subdir1").c_str()),
+              0);
+    ASSERT_EQ(
```

Comment:
We should clean this up by moving single-use files like this one into the tests using them instead of a general SetUp step. The only ones that look reused are file1 and subdir1/file2